go/lang/security/audit/crypto/missing-ssl-minversion.go

```diff
@@ -0,0 +1,70 @@
+package main
+
+import (
+	"crypto/tls"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"os"
+)
+
+// zeroSource is an io.Reader that returns an unlimited number of zero bytes.
+type zeroSource struct{}
+
+func (zeroSource) Read(b []byte) (n int, err error) {
+	for i := range b {
+		b[i] = 0
+	}
+
+	return len(b), nil
+}
+
+func main() {
+	// Dummy test HTTP server for the example with insecure random so output is
+	// reproducible.
+	server := httptest.NewUnstartedServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {}))
+	// ruleid: missing-ssl-minversion
+	server.TLS = &tls.Config{
+		Rand: zeroSource{}, // for example only; don't do this.
+	}
+	server.StartTLS()
+	defer server.Close()
+
+	// Typically the log would go to an open file:
+	// w, err := os.OpenFile("tls-secrets.txt", os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	w := os.Stdout
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			// ok: missing-ssl-minversion
+			TLSClientConfig: &tls.Config{
+				KeyLogWriter:       w,
+				MinVersion:         tls.VersionSSL30,
```

Comment:
Do we care that this is using `VersionSSL30`? I.e. is this only for insecure defaults, or also for explicitly insecure?

Reply:
This rule is for insecure defaults. We have another rule for explicitly insecure, i.e., checking for VersionSSL30.